pkg/kwokctl/utils/cmd.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+func ForkExec(ctx context.Context, dir string, name string, arg ...string) error {
+	pidPath := PathJoin(dir, "pids", filepath.Base(name)+".pid")
+	pidData, err := os.ReadFile(pidPath)
+	if err == nil {
+		_, err = strconv.Atoi(string(pidData))
+		if err == nil {
+			return nil // already running
```

Comment:
should you also call `os.FindProcess()` to ensure that it is already running?

otherwise there might be a case where if the kill failed and we do not remove the pid and that would cause inconsistencies.